### PR TITLE
Move TrackJS in order to obtain better results when any error is thrown at the beginning

### PR DESCRIPTION
**Pull request by xavijam**
Related ticket: https://github.com/CartoDB/cartodb/issues/11872.

Basically that. Where TrackJS is required, we are moving it to the top of the `<script>` files in order to follow their suggestions (http://docs.trackjs.com/tracker/installation).

Due to the fact that we don't load TrackJS in staging, we should review carefully we haven't broken anything important.

cc @CartoDB/frontend

### CODE REVIEW

**Comment by matallo** [comment]
NEWS!

**Comment by matallo** [comment]
isn't there a track.js snippet in the public dataset view? https://github.com/CartoDB/cartodb/blob/master/app/views/admin/visualizations/public_dataset.html.erb

File: app/views/admin/visualizations/embed_map_error.html.erb [hunk 1]
```diff
@@ -30,7 +30,6 @@
       </div>
     </div>
 
-    <%= insert_trackjs('embeds') %>
```
Comment (matallo) [inline app/views/admin/visualizations/embed_map_error.html.erb]: we don't have any map here, but maybe google analytics, hubspot or any other vendor scripts can throw some error?

Comment (xavijam) [inline app/views/admin/visualizations/embed_map_error.html.erb]: in the embed map error we don't provide any custom js. Take into account it is a html, just that. And if any of those fail, that's their problem, right?

**Comment by xavijam** [comment]
Further to public_dataset, is it possible that we already have it in the layout?

**Comment by matallo** [comment]
indeed, just removed my comment ><

but I couldn't find it in the dataset view i.e. "https://team.carto.com/u/matallo/dataset/untitled_table_38"

**Comment by matallo** [comment]
updated the NEWS myself, just made the comment about the track.js script missing in the dataset view, the rest looks legit 👍

**Comment by xavijam** [comment]
Let me add it in the dataset view 💋 .

**Comment by xavijam** [comment]
Thanks for NEWs update by the way.

**Comment by xavijam** [comment]
Added!

**Comment by nobuti** [comment]
Works like a charm ✨

**Pull request merged**